tests/test_artifactory.py

```diff
@@ -1,49 +1,51 @@
+import pathlib
 import sys
 import pytest
 import joblib
-from unittest.mock import patch
+from unittest.mock import MagicMock
 
 from sklearn.pipeline import Pipeline
 from sklearn.base import BaseEstimator
-from artifactory import PureArtifactoryPath
 from ml_tooling.storage.artifactory import ArtifactoryStorage
 
-
 require_artifactory = pytest.mark.skipif(
     "artifactory" not in sys.modules, reason="artifactory must be installed"
 )
 
 
 @require_artifactory
-@patch("ml_tooling.storage.artifactory.ArtifactoryPath", autospec=True)
-def test_can_load_from_artifactory(mock_artifactory_path, open_estimator_pickle):
-    mock_artifactory_path.return_value.open.return_value = open_estimator_pickle
-    f = ArtifactoryStorage("testy", "test").load("test")
+def test_can_load_from_artifactory(open_estimator_pickle):
+    mock_open = MagicMock()
+    mock_open.open.return_value.__enter__.return_value = open_estimator_pickle
+
+    mock_path = MagicMock()
+    mock_path.__truediv__.return_value.__truediv__.return_value = mock_open
+
+    storage = ArtifactoryStorage("testy", "test")
+    storage.artifactory_path = mock_path
+    f = storage.load("test")
     assert isinstance(f, (BaseEstimator, Pipeline))
 
 
 @require_artifactory
-@patch("ml_tooling.storage.artifactory.ArtifactoryPath", autospec=True)
-def test_can_save_to_artifactory(
-    mock_artifactory_path, open_estimator_pickle, tmp_path
-):
+def test_can_save_to_artifactory(open_estimator_pickle, tmp_path: pathlib.Path):
     file_path = tmp_path.joinpath("temp.pkl")
 
     def mock_deploy_file(*args, **kwargs):
-        return open(file_path, "wb").write(open_estimator_pickle.read())
+        return file_path.write_bytes(open_estimator_pickle.read())
 
-    mock_artifactory_path.return_value.deploy_file = mock_deploy_file
+    mock = MagicMock()
+    mock.__truediv__.deploy_file.return_value = mock_deploy_file()
 
     storage = ArtifactoryStorage("http://www.testy.com", "/test")
+    storage.artifactory_path = mock
     storage.save("test", "test")
     f = joblib.load(file_path)
     assert isinstance(f, (BaseEstimator, Pipeline))
```

Comment:
```suggestion
    assert isinstance(f, Estimator)
```

Reply:
Unfortunately Typehints cannot be used as instance checks